pkg/netpol/connlist/exposure_analysis_test.go

```diff
@@ -0,0 +1,192 @@
+			require.Nil(t, err, "test %q: error getting peer list", tt.testName)
+			representativePeers := pe.GetRepresentativePeersList()
+			require.Equal(t, tt.expectedNumRepresentativePeers, len(representativePeers),
+				"test %q: mismatch in number of representative peers", tt.testName)
+			peerList = append(peerList, representativePeers...)
+			peers := convertEvalPeersToConnlistPeer(peerList)
+			ca := NewConnlistAnalyzer(WithExposureAnalysis())
+			_, exposureMap, err := ca.getConnectionsBetweenPeers(pe, peers)
+			require.Nil(t, err, "test %q: error getting connections between peers", tt.testName)
+			exposedPeers := buildExposedPeerListFromExposureMap(exposureMap)
+			require.Equal(t, tt.expectedLenOfExposedPeerList, len(exposedPeers),
+				"test %q: mismatch in length of exposedPeer list", tt.testName)
+			for _, ep := range exposedPeers {
+				require.Contains(t, []string{wl1Name, wl2Name}, ep.ExposedPeer().Name(), "test: %q, unexpected exposed peer name %q",
+					tt.testName, ep.ExposedPeer().String())
+				if ep.ExposedPeer().Name() == wl1Name {
+					checkExpectedVsActualData(t, tt.testName, ep, tt.wl1ExpDataInfo)
+				} else {
+					checkExpectedVsActualData(t, tt.testName, ep, tt.wl2ExpDataInfo)
+				}
+			}
+		})
+	}
+}
+
+// todo handle error
+func buildPolicyEngineWithExposureAnalysis(t *testing.T, dirName string, numOfParsingErrs int) *eval.PolicyEngine {
+	testDir := testutils.GetTestDirPath(filepath.Join(exposureAnalysisTestsDirName, dirName))
+	rList, errs := fsscanner.GetResourceInfosFromDirPath([]string{testDir}, false, false)
+	require.Empty(t, errs, "test %q: nonempty errs list returned from the fsscanner", dirName)
+	k8sObjects, fpErrs := parser.ResourceInfoListToK8sObjectsList(rList, nil, true)
+	require.Equal(t, numOfParsingErrs, len(fpErrs),
+		"test %q: mismatch in length of errors list returned from the parser", dirName)
+	pe := eval.NewPolicyEngineWithOptions(true)
+	err := pe.AddObjects(k8sObjects)
+	require.Empty(t, err, "test %q: error adding objects to the policy engine", dirName)
+	return pe
+}
+
+func checkExpectedVsActualData(t *testing.T, testName string, actualExp ExposedPeer, expectedData expectedPeerResultInfo) {
+	require.Equal(t, expectedData.isEgressProtected, actualExp.IsProtectedByEgressNetpols(),
+		"test: %q, mismatch in is egress protected for peer %q", testName, actualExp.ExposedPeer().String())
+	require.Equal(t, expectedData.isIngressProtected, actualExp.IsProtectedByIngressNetpols(),
+		"test: %q, mismatch in is ingress protected for peer %q", testName, actualExp.ExposedPeer().String())
+	require.Equal(t, expectedData.lenIngressExposedConns, len(actualExp.IngressExposure()),
+		"test: %q, mismatch in length of ingress exposure slice for peer %q", testName, actualExp.ExposedPeer().String())
+	require.Equal(t, expectedData.lenEgressExposedConns, len(actualExp.EgressExposure()),
+		"test: %q, mismatch in length of egress exposure slice for peer %q", testName, actualExp.ExposedPeer().String())
```

Comment:
missing comparison of connection set and namespace labels?

Reply:
done